### ISSUE TITLE
Consider postcss-color-function for better color management

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5#1628]
## Is this a bug report or feature request? (choose one)

🆕 Feature request

## 📃 Other details that might be useful

Lately, I've run into a problem of styling a considerable amount of elements with the same main color, but with different lightness/opacity. Well, we use variables, but it still meant that I need to define 8+ variables and use each of them just once (will be even more variables in the future and there are already several using that main color, so it is not a problem of 2-3 variables).

It also makes it a nightmare if we (or integrator) would like to change the styling, as you need to change all of them.

What would be best is this:

```
--ck-my-feature-my-color: hsla( 128, 70%, 50%, 0.5 );

/* ...  */

background-color: lighten( --ck-my-feature-my-color, 0.1 );
```

At the moment I could go with:
```
--ck-my-feature-my-color-hsl: 120, 70%, 50%;

/* ...  */

background-color: hsla( var( --ck-my-feature-my-color-hsl ), 0.4 );
```

This is better than having single variable for every value but still not flexible and not prefect.

**Comment by Reinmar** [comment]
cc @oleq (we need some opinion here :D)

**Comment by oleq** [comment]
We used to have https://github.com/postcss/postcss-color-function in the past and we dropped it. The thing was there was something wrong with it and I cannot put my finger on it now because it was a long time ago.

We could give it a try. But that means some research.

**Comment by oleq** [comment]
Ah, I guess I remember now. 

To do the thing like `lighten( var( --some-color ), 10% )` (or whatever), the plugin [requires the `postcss-custom-properties` plugin](https://github.com/postcss/postcss-color-function#can-we-use-css-custom-properties-so-we-can-do-colorvar--maincolor-adjuster) because the value of `--some-color` is determined only during the **browser** runtime and rendering.

What `postcss-custom-properties` does is parsing all the custom props when the styles are built (postcss running) so they are available as a data structure for other plugins (like postcss-color-function) to read from. 

AFAIR there were some problems with custom-props plugin:
* when used it completely consumed the `:root { --some-color: ... }` declarations in the postcss output. So it stripped the application from custom CSS properties, so no more custom themes for CKEditor using variables, no easy customizations, etc.
* it duplicated some CSS (don't remember which exactly, but it was a problem)
* it was incompatible with our styles building system (core styles and theme styles on top, loaded by our custom postcss plugin); the custom props were lost or not recognized or whatever, but there were issues

So long story short

> What would be best is this:

Nope :P
